### ISSUE TITLE
Use visual nudges to encourage people to write more in their Notes

### ISSUE THREAD

**Issue by RobJN** [openstreetmap/openstreetmap-website#2415]
Some OSM Notes are simply not long enough to be of use. We don't need an essay but a little more detail would help. 

One idea is to use a visual nudge to encourage people to write a bit more. I recently saw this when reviewing a hotel on booking.com. They provide a sort of progress bar which starts out at a low level and coloured red, and then increases as you type more until it is green and full. You also get a small bit of text encouraging you to write more, followed by "nearly there" etc.

Similar visual nudges exist when having to set a new password with a status bar in red for "Weak" passwords, orange for "OK" passwords and green for "Great" passwords.

What are you views on implementing similar for OSM Notes?

**Comment by tomhughes** [comment]
Well the obvious problem is determining what amount of text is "good" without being able to read the users mind - sometimes a few words is fine and other times it isn't.

A hotel review is rather different because it's obvious that you can't provide a meaningful review that is only half a dozen words, but "this road is closed" might be all that is needed in a note.

**Comment by simonpoole** [comment]
Do we have any idea how many notes are created via the website vs other apps using the api.

**Comment by RobJN** [comment]
Yes I appreciate it is not easy to know the right length but I still think there is value in encouraging a minimal amount of text. 

For your example of "this road is closed" there is a good chance that more information is required to update the map. Is the closure permanent? how much of the road is closed (e.g. between junction X and junction Y)? And so on. I certainly would like a bit more info before making a map edit.

In the absence of any stats/evidence, may I start by suggesting a length of 50 characters as the level where the nudge indicator goes green. Uses can of course add more or less as they see fit.

**Comment by mmd-osm** [comment]
Before spending time on some fancy javascript code with unclear results, let's review what we have today:

1. The placeholder text   _"Your note is public and may be used to update the map, so don't enter personal information, or information from copyrighted maps or directory listings."_ doesn't provide much guidance towards the level of detail which we expect a user to provide.

2. _" and type a note to explain the problem."_ also doesn't encourage people to provide a detailed report. 

Could those be reworded/improved somehow?

**Comment by RobJN** [comment]
Good idea. How about:

A good note will allow others to update the map. Please be as descriptive as possible but avoid entering personal information, or information from copyrighted maps or directory listings.

**Comment by mmd-osm** [comment]
+ "Please be nice, patient and respectful when communicating with our community members."

**Comment by Fizzie41** [comment]
> 1. The placeholder text   _"Your note is public and may be used to update the map, so don't enter personal information, or information from copyrighted maps or directory listings."_ doesn't provide much guidance towards the level of detail which we expect a user to provide.

On this, DWG have just had an issue raised by somebody who (despite this ^ !) thought they were leaving a private comment that no-one else could see.

Their suggestion: "_It would be useful to add the word public in the context menu so instead of just saying "Add a note here" it says something like: "Add a note here (Public)" or "Add a public note here".

Reasoning: the moment I start typing in the text box for the note the placeholder text vanishes, since that is the only warning about it being public it can be easily missed by someone who wasnt looking for it or just saw the text box and started typing ignoring the placeholder text inside as I did._"

Two thoughts following on from this.

Maybe change the working to "Add a public Note, visible to everybody, here"?

And possibly also add an acknowledgement button, so that the placeholder text stays there, until you click to say you're read it?

**Comment by 1ec5** [comment]
> It would be useful to add the word public in the context menu so instead of just saying "Add a note here" it says something like: "Add a note here (Public)" or "Add a public note here".

Adding “public” somewhere in the note panel seems reasonable to me, since “note” is such an overloaded term even within OSM. (Edit: As noted below, it probably wouldn’t have much effect in the context menu itself, since that’s so fleeting.)

> the moment I start typing in the text box for the note the placeholder text vanishes

What if we move the placeholder text outside of the textbox, so it remains visible even after the user begins typing? It would take up more space, but in general we probably can’t expect users to closely read long placeholder text anyways.

**Comment by tomhughes** [comment]
That's certainly better than adding something to the menu which is both ugly and probably won't work as people will forget it once they've clicked on it just as with the placeholder.

**Comment by kmpoppe** [comment]
> Do we have any idea how many notes are created via the website vs other apps using the api?

I can now answer this from guessing the app that created note by painful text analysis (see [here](https://dupnotes.webmapping.cyou)), ⅔ of the notes currently are considered to be from the website or via the API without announcing a specific app.

Thus, imho, having an encouraging text in the New Notes screen isn't wasted.

I'd like to ask people to look at the PR attached here, if they like the solution that's on the table now. Thanks :)
